frontends/PyCDE/src/pycde/bsp/Makefile.xrt.j2

```diff
@@ -0,0 +1,160 @@
+
+# Package everything into a Vitis compatible kernel (.xo format)
+$(XO_OUT): $(TEMP)
+	$(VIVADO) -mode batch -source $(SRC)/xrt_package.tcl -tclargs $(SRC) $(XO_OUT) $(TARGET) $(PLATFORM) $(XSA)
+
+# Link Vitis system using the generated kernel for the chosen platform
+$(LINK_OUT): $(XO_OUT) | $(BUILD)
+	$(VPP) $(VPPFLAGS) -l --temp_dir $(TEMP) -o'$(LINK_OUT)' $(+)
+
+# Build the xclbin
+$(XCL_OUT): $(LINK_OUT)
+	$(VPP) -p $(LINK_OUT) $(VPPFLAGS) --package.out_dir $(PACKAGE) -o $(XCL_OUT)
+
+# Generate configuration for use with hw_emu mode
+emconfig: $(BUILD)/emconfig.json
+$(BUILD)/emconfig.json:
+	emconfigutil --platform $(PLATFORM) --od $(BUILD)
+
+# Compile the Python interface driver.
+# TODO: build for a list of python versions.
+PY_EXT := $(shell $(PYTHON)-config --extension-suffix)
+PYBIND11_INC := $(shell $(PYTHON) -m pybind11 --includes)
+runtime/$(NAME)/esiXrtPython$(PY_EXT): runtime/$(NAME)/EsiXrtPython.cpp
+	$(CXX) -o runtime/$(NAME)/esiXrtPython$(PY_EXT) $^ $(CXXFLAGS) $(LDFLAGS) -shared -fPIC $(PYBIND11_INC)
+esiXrtPython: runtime/$(NAME)/esiXrtPython$(PY_EXT)
+
+clean:
+	rm -rf $(BUILD) .Xil vivado* kernel *.jou *.log *.wdb *.wcfg *.protoinst *.csv
+	rm runtime/*.so
+
+# Targets which only apply to image builds.
+ifeq ($(TARGET), hw)
+
+# Submit the image to Azure for attestation. Follows the instructions at:
+# https://learn.microsoft.com/en-us/azure/virtual-machines/field-programmable-gate-arrays-attestation
+IMAGE_AZ_BASENAME ?= $(NAME)_$(shell date +%s).hw
+IMAGE_AZ_NAME := $(USER)_$(IMAGE_AZ_BASENAME)
+azure: $(IMAGE_AZ_NAME).azure.xclbin
+azure_creds:
+	@echo "*************************"
+	@echo "* Getting Azure credentials. MUST 'az login' first!"
+	@echo "*************************"
+	@ if [ "${AZ_FPGA_SUB}" = "" ] || [ "${AZ_FPGA_STORAGE_ACCOUNT}" = "" ] || \
+	     [ "${AZ_FPGA_STORAGE_CONTAINER}" = "" ]; then \
+		@echo "** AZ_FPGA_SUB, AZ_FPGA_STORAGE_ACCOUNT, and AZ_FPGA_STORAGE_CONTAINER" \
+		exit 1; \
+	fi
+
+	$(eval SAS_EXPIRY=$(shell date --date "now + 48hours" +"%Y-%m-%dT%0k:%MZ"))
+	$(eval SAS=$(shell \
+		az storage container generate-sas \
+			--subscription $(AZ_FPGA_SUB) \
+			--account-name $(AZ_FPGA_STORAGE_ACCOUNT) \
+			--name $(AZ_FPGA_STORAGE_CONTAINER) \
+			--https-only --permissions rwc --auth-mode key \
+			--expiry $(SAS_EXPIRY) --output tsv))
+
+$(IMAGE_AZ_NAME).azure.xclbin: azure_creds $(XCL_OUT) validate-fpgaimage.sh
+	@echo "*************************"
+	@echo "* Submitting job to Azure attestation."
+	@echo "* This step WILL take a LONG time (between 30 mins and 1.5 hours)."
+	@echo "*   Using name $(IMAGE_AZ_NAME)"
+	@echo "*************************"
+
+	az storage blob upload \
+		--subscription $(AZ_FPGA_SUB) \
+		--account-name $(AZ_FPGA_STORAGE_ACCOUNT) \
+		--container-name $(AZ_FPGA_STORAGE_CONTAINER) \
+	  --sas-token "$(SAS)" --overwrite \
+		--name $(IMAGE_AZ_NAME).xclbin --file $(NAME).hw.xclbin
+
+	bash validate-fpgaimage.sh --storage-account $(AZ_FPGA_STORAGE_ACCOUNT) \
```

Comment:
Nit : formatting

Reply:
This is a result of tabs vs. spaces. I use two spaces for tabs whereas GH displays 4. Tabs... *sigh*.